### PR TITLE
MueLu: Material Aggregation (WIP)

### PR DESCRIPTION
**Pull request by csiefer2**
This code does not yet pass all unit tests.

Somehow I broke BlockedRepartition's BlockedRAPFactoryWithDiagonal unit test and I could use an assist figuring out how.

Thanks!

### CODE REVIEW

File: packages/muelu/test/unit_tests/BlockedRepartition.cpp [hunk 1]
```diff
@@ -886,7 +886,7 @@ namespace MueLuTests {
   RCP<RepartitionHeuristicFactory> RepHeuFact = Teuchos::rcp(new RepartitionHeuristicFactory);
   RepHeuFact->SetFactory("A",AcFact);//MueLu::NoFactory::getRCP());
   RepHeuFact->SetParameter("repartition: start level",Teuchos::ParameterEntry(0));
-  RepHeuFact->SetParameter("repartition: min rows per proc",Teuchos::ParameterEntry(40));
+  RepHeuFact->SetParameter("repartition: min rows per proc",Teuchos::ParameterEntry(100));
```
Comment (cgcgcg) [inline packages/muelu/test/unit_tests/BlockedRepartition.cpp]: Why did this need to be changed?

Comment (csiefer2) [reply packages/muelu/test/unit_tests/BlockedRepartition.cpp]: It didn't.  I just wanted to trigger the issue on 2 ranks rather than 4.

Comment (csiefer2) [reply packages/muelu/test/unit_tests/BlockedRepartition.cpp]: For debugging purposes.

**Comment by jhux2** [comment]
@csiefer2 This error is reproducible on two processors.  One of the processors is hanging in MueLu_CoordinatesTransferFactory_def.hpp:224, on a call to MapFactory::Build().  The other is running on to an MPI all reduce in MueLu_Hierarchy_def.hpp:1024 in Hierarchy::describe().

**Comment by jhux2** [comment]
```
mpirun -np 2 MueLu_UnitTests_Blocked.exe --group=BlockedRepartition_double_int_int_Kokkos_Compat_KokkosSerialWrapperNode --test=BlockedRAPFactoryWithDiagonal  --pause-for-debugging
```

**Pull request closed without merging**